cpp/rn-llama.cpp

```diff
@@ -719,4 +725,101 @@ std::vector<common_adapter_lora_info> llama_rn_context::getLoadedLoraAdapters()
     return this->lora;
 }
 
+bool llama_rn_context::initMultimodal(const std::string &mmproj_path) {
+    if (model == nullptr) {
+        LOG_ERROR("Model not loaded, cannot initialize multimodal", "");
+        return false;
+    }
+
+    // Initialize mtmd context
+    mtmd_context_params mtmd_params = mtmd_context_params_default();
+    mtmd_params.use_gpu = true;
+    mtmd_params.print_timings = true;
+    mtmd_params.n_threads = params.cpuparams.n_threads;
+    mtmd_params.verbosity = (lm_ggml_log_level)LM_GGML_LOG_LEVEL_INFO;
+
+    mtmd_ctx = mtmd_init_from_file(mmproj_path.c_str(), model, mtmd_params);
+    if (mtmd_ctx == nullptr) {
+        LOG_ERROR("Failed to initialize multimodal context with mmproj: %s", mmproj_path.c_str());
+        return false;
+    }
+
+    has_multimodal = true;
+    LOG_INFO("Multimodal context initialized successfully with mmproj: %s", mmproj_path.c_str());
+    return true;
+}
+
+bool llama_rn_context::processImage(
+    const std::string &image_path,
+    const std::string &prompt
+) {
+    if (!has_multimodal || mtmd_ctx == nullptr) {
+        LOG_ERROR("Multimodal context not initialized", "");
+        return false;
+    }
+
+    // Load the image from file
+    mtmd_bitmap* bitmap = mtmd_helper_bitmap_init_from_file(image_path.c_str());
+    if (bitmap == nullptr) {
+        LOG_ERROR("Failed to load image: %s", image_path.c_str());
+        return false;
+    }
+
+    // Create input chunks
+    mtmd_input_chunks* chunks = mtmd_input_chunks_init();
+    if (chunks == nullptr) {
+        LOG_ERROR("Failed to initialize input chunks", "");
+        mtmd_bitmap_free(bitmap);
+        return false;
+    }
+
+    // Create input text
+    mtmd_input_text input_text;
+    input_text.text = prompt.c_str();
+    input_text.add_special = true;
+    input_text.parse_special = true;
+
+    // Tokenize the text and image
+    const mtmd_bitmap* bitmaps[] = { bitmap };
+    int32_t res = mtmd_tokenize(mtmd_ctx, chunks, &input_text, bitmaps, 1);
+    if (res != 0) {
+        LOG_ERROR("Failed to tokenize image and text: %d", res);
+        mtmd_input_chunks_free(chunks);
+        mtmd_bitmap_free(bitmap);
+        return false;
+    }
+
+    // Evaluate the chunks in the model's context
+    // This is the critical step that makes the model aware of the image
+    llama_pos new_n_past = 0;
+    if (mtmd_helper_eval_chunks(mtmd_ctx,
+                ctx, // llama context
+                chunks, // chunks
+                0, // n_past (start from beginning)
```

Comment:
Will it works if we add n_past here:

```suggestion
                n_past, // n_past (start from beginning)
```